src/cascade/testing_utilities/dtypes.py

```diff
@@ -0,0 +1,20 @@
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+
+
+def print_types(data):
+    """Walks through data types and prints them out.
+
+    >>> input_data = retrieve_data(ec, local_settings, covariate_data_spec)
+    >>> print_types(input_data)
+
+    """
+    with Path(f"dtypes{np.random.randint(0, 999)}.txt").open("w") as outfile:
+        for member_name in dir(data):
+            member = getattr(data, member_name)
+            if isinstance(member, pd.DataFrame):
+                dtypes = member.dtypes
+                as_data = list(zip(dtypes.index.tolist(), dtypes.tolist()))
+                print(f"data_type {member_name}: {as_data}", file=outfile)
```

Comment:
Should this be a print or a log?

Reply:
erm. That's under testing. I like the print there. Prints are bad, though... I'll change it.

Reply:
Wait. That's going to a file, on purpose. That function was there to snoop on what comes in. I used it this way: I'd get stuff from the IHME db, not know what the columns are, and use this to see if they varied... You know, I should do that with the debugger... aaaaand deleted the file.

Reply:
Oh, I didn't even notice it was going to a file. But moot point I guess.